src/org/zaproxy/zap/extension/groovy/files/scripts/templates/variant/InputVectorDefaultTemplate.groovy

```diff
@@ -0,0 +1,79 @@
+import groovy.transform.Field
+import org.apache.commons.lang.StringUtils
+import org.parosproxy.paros.core.scanner.NameValuePair
+import org.parosproxy.paros.core.scanner.VariantCustom
+import org.parosproxy.paros.network.HttpMessage
+
+import java.nio.charset.StandardCharsets
+
+/*
+A script that provides custom input vectors that can be used during active
+scans.
+The option Enable Script Input Vectors must be enabled before starting the
+scans.
+Note that new scripts will initially be disabled, right click the script in the
+Scripts tree and select Enable Script.
+*/
+
+@Field final String CUSTOM_HEADER = 'My-Custom-Header'
+
+/*
+Parses/extracts the parameters from the HTTP message (request), to be
+later tested by the scanners.
+Args:
+    helper (VariantCustom): Helper class that provides functions to add new
+                            parameters and process its values.
+    msg (HttpMessage):      The HTTP message (request) that will be scanned.
+*/
+void parseParameters(VariantCustom helper, HttpMessage msg){
+
+    //Extract the attributes of a custom header...
+    def header = msg.getRequestHeader().getHeader(CUSTOM_HEADER)
+    if (header == null){
+        return
+    }
+
+    def attributes = header.split(';')
+    for (def attribute : attributes){
+        if(StringUtils.isEmpty(attribute)){
+            continue
+        }
+
+        def data = attribute.split('=')
+        if(data.length != 2){
+            continue
+        }
+
+        def name = data[0]
+        def value = URLDecoder.decode(data[1], StandardCharsets.ISO_8859_1)
```

Comment:
This fails with:
```
java.lang.reflect.UndeclaredThrowableException
	at com.sun.proxy.$Proxy38.parseParameters(Unknown Source)
	at org.parosproxy.paros.core.scanner.VariantCustom.setMessage(VariantCustom.java:81)
	at org.parosproxy.paros.core.scanner.AbstractAppParamPlugin.scan(AbstractAppParamPlugin.java:161)
	at org.parosproxy.paros.core.scanner.AbstractPlugin.run(AbstractPlugin.java:339)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoSuchMethodException: No signature of method: static java.net.URLDecoder.decode() is applicable for argument types: (java.lang.String, sun.nio.cs.ISO_8859_1) values: [b, ISO-8859-1]
Possible solutions: decode(java.lang.String), decode(java.lang.String, java.lang.String), hashCode()
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.invokeImpl(GroovyScriptEngineImpl.java:364)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.access$200(GroovyScriptEngineImpl.java:94)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl$4.invoke(GroovyScriptEngineImpl.java:407)
	... 5 more
```
should be `StandardCharsets.ISO_8859_1.name()`?

Reply:
Good catch, thank you.